src/components/organisms/Badges/Badges.tsx

```diff
@@ -1,64 +1,81 @@
-import React, { useMemo } from "react";
+import React, {
+  RefObject,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
 import { User } from "@bugsnag/js";
 
 import { Link } from "react-router-dom";
-import { useFirestoreConnect } from "react-redux-firebase";
-
-import { useConnectRelatedVenues } from "hooks/useConnectRelatedVenues";
-import { useSelector } from "hooks/useSelector";
+import { useFirestore } from "react-redux-firebase";
 
 import { PLAYA_VENUE_NAME } from "settings";
 
 import { CampRoomData } from "types/CampRoomData";
 import { isVenueWithRooms } from "types/CampVenue";
-import { AnyVenue } from "types/Firestore";
+import { AnyVenue, UserVisit } from "types/Firestore";
 import { Venue } from "types/Venue";
 
 import { WithId } from "utils/id";
-import {
-  orderedVenuesSelector,
-  userModalVisitsSelector,
-} from "utils/selectors";
 import { venueInsideUrl, venuePreviewUrl } from "utils/url";
 import { notEmpty } from "utils/types";
 
 export const Badges: React.FC<{
   user: WithId<User>;
   currentVenue: WithId<Venue>;
 }> = ({ user, currentVenue }) => {
-  useFirestoreConnect([
-    {
-      collection: "users",
-      doc: user.id,
-      subcollections: [{ collection: "visits" }],
-      storeAs: "userModalVisits",
-    },
-  ]);
-  useFirestoreConnect("venues");
-  const visits = useSelector(userModalVisitsSelector);
-  const venues = useSelector(orderedVenuesSelector);
-
-  const { relatedVenues } = useConnectRelatedVenues({
-    venueId: currentVenue.id,
-    withEvents: false,
-  });
-  const relatedVenueNames = relatedVenues.map((venue) => venue.name);
+  const [visits, setVisits] = useState<WithId<UserVisit>[]>([]);
+  const [venues, setVenues] = useState<WithId<AnyVenue>[]>([]);
+
+  const imgRef: RefObject<HTMLImageElement[]> = useRef([]);
+
+  const firestore = useFirestore();
+
+  const fetchAllVenues = useCallback(async () => {
+    const venuesSnapshot = await firestore.collection("venues").get();
+
+    const userSnapshot = await firestore.collection("users").doc(user.id).get();
+    const visitsSnapshot = await userSnapshot.ref.collection("visits").get();
+
+    const venues =
+      venuesSnapshot.docs.map(
+        (venueSnapshot) =>
+          ({ ...venueSnapshot.data(), id: venueSnapshot.id } as WithId<
+            AnyVenue
+          >)
```

Comment:
I believe the `as` is being used here because firebase uses `DocumentData` (or similar) as it's type. Does that take a generic parameter where we could pass this type without needing to explicitly cast it here?

Reply:
I looked around but there doesn't seem to be an option for this.